buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/fixtures/IntTestImagePlugin.kt

```diff
@@ -122,10 +122,16 @@ open class IntTestImagePlugin : Plugin<Project> {
         // TODO: Model these as publications of different types of distributions, collapse the number of variants exposed by the root project
         // and eliminate duplication with distributions.gradle
         dependencies {
-            gradleRuntimeSource(project(":"))
-            gradleRuntimeSource(project(":apiMetadata"))
-            gradleDocumentation(project(":docs"))
-            gradleScripts(project(":launcher"))
+            gradleRuntimeSource.withDependencies {
```

Comment:
This config code is now also executed as part of pre-compiled script compilation. There the projects we try to find here (e.g. `project(":apiMetadata")`) do not exist. We now only apply the dependencies once the configuration is resolved (`withDependencies` hook).